### ISSUE TITLE
RGB: Make def image_data More Efficient with Big O

### ISSUE THREAD

**Issue by CalrethilOfMirkwood** [CalrethilOfMirkwood/flask_portfolio#15]
Image files are really big, as programmers how could we make processing them.  Could the image.py function "def image_data" more efficient?  There is a science to writing efficient algorithms called Big O notation. Write image_data function to be more efficient according to Big O notation.

Results 
I have managed to make the function work more efficiently. If I used Big O notation, the efficiency of the original function was O(n*m). The difference between the old function and the new function is that the old function calculated every single pixel even if it was the same as used before. The new function that I was able to create memorizes the data and so if the same RGB values were inputted in, then the computer wouldn't need to calculate it again. It instead can just take the memorized values and use that. Due to this code this means that instead of M the efficiency will be a constant. Therefore the efficiency of the new function is O(n). The proof of the code and pictures of the code can be found on my journal. https://docs.google.com/document/d/11vsWPaCuoXcu6-stnnRsQocQagZt6etMUKN_my-wCZQ/edit

**Comment by FierceExplorer** [comment]
I have managed to make the function work more efficiently. If I used Big O notation, the efficiency of the original function was O(n*m). The difference between the old function and the new function is that the old function calculated every single pixel even if it was the same as used before. The new function that I was able to create memorizes the data and so if the same RGB values were inputted in, then the computer wouldn't need to calculate it again. It instead can just take the memorized values and use that. Due to this code this means that instead of M the efficiency will be a constant. Therefore the efficiency of the new function is O(n). The proof of the code and pictures of the code can be found on my journal. https://docs.google.com/document/d/11vsWPaCuoXcu6-stnnRsQocQagZt6etMUKN_my-wCZQ/edit
